gcloud-java-core/src/main/java/com/google/gcloud/Service.java

```diff
@@ -16,6 +16,6 @@
 
 package com.google.gcloud;
 
-public interface Service<O extends ServiceOptions> {
-  O options();
+public interface Service<OptionsT extends ServiceOptions<?, OptionsT>> {
```

Comment:
I think Google's Java style is much closer to widely-accepted best practices than Google's Python style. It is true that the `SomethingT` convention specifically is something of an innovation, but one where the advantages over the single-letter convention are pretty plain to see. I don't think `SomethingType` is any better established, and it is much more ambiguous.